src/lxc/attach.c

```diff
@@ -250,8 +250,7 @@ static int lxc_attach_to_ns(pid_t pid, int which)
 			continue;
 		}
 
-		snprintf(path, MAXPATHLEN, "/proc/%d/ns/%s", pid, ns[i]);
-		fd[i] = open(path, O_RDONLY | O_CLOEXEC);
+		fd[i] = lxc_preserve_ns(pid, ns[i]);
 		if (fd[i] < 0) {
 			saved_errno = errno;
```

Comment:
In my tree, below this there is a SYSERROR("failed to open '%s'", path); below this which you seem to be leaving in place, but path is no longer the right name.